### PR TITLE
Understanding Docker Networking

### PR DESCRIPTION
**Pull request by Pauline-Mwangi**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by Pauline-Mwangi** [comment]
This fixes #2396

**Comment by ahmadmardeni1** [comment]
Any update on this PR? @cyberShaw

**Comment by cyberShaw** [comment]
> Any update on this PR? @cyberShaw 

I'll do the review shortly

**Comment by ahmadmardeni1** [comment]
Can you please review this one? @WanjaMIKE

**Comment by ahmadmardeni1** [comment]
Sorry for the waiting, we will review it Asap @Pauline-Mwangi

File: content/articles/understanding-docker-networking/index.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+[Docker Networking](#docker-networking)
+[Why Docker Networking is Important](#why-docker-networking-is-important)
+[Container Network Models](#container-network-models)
+[Network Drivers](#network-drivers)
+
+### Prerequisites
+A comprehensive understanding of [Docker](https://www.section.io/engineering-education/docker-concepts/) is needed to follow along with this article. Also, go through [networking overview](https://docs.docker.com/network/) on Docker documentation.
+
+### Docker networking
+Networking involves communication among processes. Networking in Docker is no different. Docker networking initiate communication between Docker containers and the outside world via the [host machine](https://docs.docker.com/network/host/). Docker utilizes a container network model (CNM) for networking. This blueprint regulates the steps needed to provide networking for containers using drivers.
```
Comment (WanjaMIKE) [inline content/articles/understanding-docker-networking/index.md:17]: Could you elaborate more on this sentence `Networking involves communication among processes`

File: content/articles/understanding-docker-networking/index.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+The application does not work on another system because there is a difference in the computer systems. Also, it may be because of differences in hardware and software. The differences are possible in two different machines. The ideal solution to this problem is Docker networking.
+
+Docker networking enables containers to talk to other containers, and once applied, it should run applications across different systems. With Docker networking, the application works fine and can work on any system. Docker networking is a crucial concept when different containers run in various networks. So it resolves most problems. You can have multiple containers running in the same network or particular networks.
+
+### Container network models (CNM)
+CNM is a [libnetwork](https://github.com/moby/libnetwork). Docker and the Docker community drive it. CNM introduces interoperation between networks. It conceptualizes the steps needed to provide networking for containers.
```
Comment (WanjaMIKE) [inline content/articles/understanding-docker-networking/index.md:35]: Could elaborate more on what `libnetwork` entails

**Review by WanjaMIKE — CHANGES_REQUESTED** [review]
This appears to be a `well-written` article. Could you go through the above minor suggestions. Thanks

**Comment by hectorkambow** [comment]
[Understanding Docker Networking #2570.pdf](https://github.com/section-engineering-education/engineering-education/files/6727580/Understanding.Docker.Networking.2570.pdf)

@Pauline-Mwangi 
👋 Good afternoon - and welcome to the EngEd program.
I'll be running a final review on your PR when ready.
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

**Comment by WanjaMIKE** [comment]
@Pauline-Mwangi also correct the plagiarism before we move forward. Thanks

**Comment by WanjaMIKE** [comment]
@Pauline-Mwangi any updates?

**Comment by Pauline-Mwangi** [comment]
@WanjaMIKE I made the changes you suggested and worked on plagiarism

**Comment by WanjaMIKE** [comment]
Checking the changes ...

**Review by WanjaMIKE — APPROVED** [review]
In the next article, work on your flow. Avoid `"sudden"` sentences that distract the reader.

**Comment by ahmadmardeni1** [comment]
Good work ladies 👍🚀 @Pauline-Mwangi @WanjaMIKE

**Pull request merged**